interoperator/pkg/internal/config/config.go

```diff
@@ -0,0 +1,108 @@
+package config
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/constants"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("config.manager")
+
+// InteroperatorConfig contains tuneable configs used by interoperator
+type InteroperatorConfig struct {
+	InstanceWorkerCount int `yaml:"instanceWorkerCount,omitempty"`
+	BindingWorkerCount  int `yaml:"bindingWorkerCount,omitempty"`
+}
+
+// newInteroperatorConfig assigns default values to config
+func newInteroperatorConfig() *InteroperatorConfig {
+	return &InteroperatorConfig{
+		BindingWorkerCount:  constants.DefaultBindingWorkerCount,
+		InstanceWorkerCount: constants.DefaultInstanceWorkerCount,
+	}
+}
+
+// Config fetches the runtime configs from the configmap
+type Config interface {
+	GetConfig() *InteroperatorConfig
+}
+
+type config struct {
+	kubeConfig *rest.Config
+	clientset  *kubernetes.Clientset
+	configMap  *corev1.ConfigMap
+}
+
+// New returns a new Config using the kubernetes client
+func New(kubeConfig *rest.Config) (Config, error) {
+	if kubeConfig == nil {
+		return nil, fmt.Errorf("invalid input to new config")
+	}
+	// create the clientset
+	clientset, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+	return &config{
+		kubeConfig: kubeConfig,
+		clientset:  clientset,
+	}, nil
+}
+
+func (cfg *config) updateConfig() error {
+	var err error
+	cfg.configMap, err = cfg.clientset.CoreV1().
+		ConfigMaps(constants.ServiceFabrikNamespace).
+		Get(constants.ConfigMapName, metav1.GetOptions{})
```

Comment:
Function name is confusing, why name it `updateConfig` when we are fetching config?

Reply:
Renamed